internal/graph/check.go

```diff
@@ -624,6 +651,56 @@ func (cc *ConcurrentChecker) checkComputedUserset(ctx context.Context, crc curre
 	return combineResultWithFoundResources(result, membershipSet)
 }
 
+// queryOptionsForArrowRelation returns query options such as SkipCaveats and SkipExpiration if *none* of the subject
+// types of the given relation support caveats or expiration.
+func (cc *ConcurrentChecker) queryOptionsForArrowRelation(ctx context.Context, reader datastore.Reader, namespaceName string, relationName string) ([]options.QueryOptionsOption, error) {
+	// TODO(jschorr): Change to use the type system once we wire it through Check dispatch.
+	nsDefs, err := reader.LookupNamespacesWithNames(ctx, []string{namespaceName})
```

Comment:
Wouldn't this add a new roundtrip in the critical path? 😬 Any potential gains by eliding columns would be thwarted by this new extra query.

Reply:
It will be cached since the namespace was already accessed for the check previously

Reply:
ah, you are right